internal/pkg/call/call_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package call
+
+import (
+	"fmt"
+	"go/ast"
+	"go/importer"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/tools/go/ssa"
+	"golang.org/x/tools/go/ssa/ssautil"
+)
+
+type fakeReferrer struct {
+	hasPathTo bool
+}
+
+func (f fakeReferrer) RefersTo(node ssa.Node) bool { return f.hasPathTo }
+
+func TestRegularCallReferredBy(t *testing.T) {
+	source := readFromTestData("test.go")
+	cases := []struct {
+		r              Referrer
+		parentFuncName string
+		want           bool
+	}{
+		{fakeReferrer{true}, "CallNoArgs", false},
+		{fakeReferrer{true}, "CallOneArg", true},
+		{fakeReferrer{false}, "CallOneArg", false},
+	}
+	for _, c := range cases {
+		ssaCall := getCall(source, c.parentFuncName)
+		call := Regular(ssaCall)
+		got := call.ReferredBy(c.r)
+		if got != c.want {
+			t.Errorf("call.ReferredBy(%v) == %v, want %v", call, got, c.want)
+		}
+	}
+}
+
+func readFromTestData(filename string) string {
+	filePath := filepath.Join("testdata", filename)
+	bytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		panic(err)
+	}
+	return string(bytes)
+}
+
+// taken from golang.org/x/tools/go/ssa/example_test.go
+func getCall(source, parentFuncName string) *ssa.Call {
+	fset := token.NewFileSet()
```

Comment:
This is very cool.

However, I want us to standardize on a single approach for extracting things from testdata.
Up until this point we have been leveraging `&analysis.Analyzer` to extract interesting things for testing.

For example:
https://github.com/google/go-flow-levee/blob/master/internal/pkg/source/source_test.go#L63

Do you think we could use `&analysis.Analyzer` here is well?

Reply:
I think it would be possible to use `&analysis.Analyzer` + `analysistest`, but in this case I would prefer the unit test. I think turning this into an analyzer would make the code significantly more complex than it needs to be.

Reply:
> I think turning this into an analyzer would make the code significantly more complex than it needs to be.

I would disagree with this. I think that using `analysistest` would simplify the logic of testing because:
1. `readFromTestData` would go away
2. `analysistest` hides the details of how SSA components (ex. calls) are extracted from sources
3. (for better of worse) we already standardized on the approach of `analysistest`.

Reply:
I agree that the *testing* code would be simpler, but the code for handling calls would be more complex (in my opinion). I think you raise important points though. I'll give it a shot.